packages/vue-components/components/commons/generic-form/generic-form.vue

```diff
@@ -0,0 +1,193 @@
+<template>
+  <PrimeMessage v-if="fields.length === 0" severity="error">
+    No fields to display. Please configure correctly your form.
+  </PrimeMessage>
```

Comment:
Should we handle that kind of "development error" this way?

Perhaps this is a usual practice in the "front-end" world?

Because I would rather expect an exception.

Reply:
well, we can't throw errors in the HTML :-)

VueJS knows the `fields` property is mandatory/required. In development, a warning will be displayed to the developer in the console, but when built, ... well, nothing happens. 

I think it will be above, in the parent component, that we'll handle this case where the form has no field.

But... for other developers, maybe it could be nice that the generic form display that kind of message, to help ?

What do you think ?

Reply:
I think errors done by developers should not end in such messages addressed to themselves through the application GUI as if they were final users.

Is there any component _event_ or _hook_ from which we could check that the `fields` property is not empty and throw an exception if not? Exception that should be handled at an upper level of the application and lead, at this moment, to a generic technical error message like "A technical error occurred..." displayed to the user.

Reply:
This generic error message may contain the initial error message during development.

Reply:
We could throw an error on the `onMounted` hook, or even in the setup, but... that's not the way I think we have to handle those cases. 
Throwing errors could break the SPA ? (I suppose, in fact, as I never did this before)

That's why we set the `fields` property to an empty array, to avoid breaking the component.

So, maybe the best thing to do here is to not display this message.

As it is typed, developer using Volar extension will see an issue in its code. No more.

Are you ok with this ?

Reply:
> So, maybe the best thing to do here is to not display this message.
> As it is typed, developer using Volar extension will see an issue in its code. No more.

Ok for that! :slightly_smiling_face: